bundles/mapping/mapmodule/plugin/togglenavigation/ToggleNavigation.js

```diff
@@ -0,0 +1,175 @@
+            const nav = [...Oskari.dom.getRootEl().children].find(c => c.localName === 'nav');
+            if (!el || !nav) {
+                return;
+            }
+            const isToggled = !!this.state.navigationVisible;
+            ReactDOM.render(
+                <StyledButton
+                    className='t_navigationtoggle'
+                    visible={this.isVisible()}
+                    icon={<MenuOutlined />}
+                    iconActive={isToggled}
+                    onClick={() => {
+                        this.setState({
+                            navigationVisible: !isToggled
+                        });
+                    }}
+                    position={this.getLocation()}
+                />
+                ,
+                el[0]
+            );
+        },
+        teardownUI: function () {
+            this.removeFromPluginContainer(this.getElement());
+        },
+        /**
+         * @method _stopPluginImpl
+         *
+         * @param {Oskari.mapframework.sandbox.Sandbox} sandbox
+         *          reference to application sandbox
+         */
+        _stopPluginImpl: function (sandbox) {
+            this.teardownUI();
+        },
+
+        /**
+         * @private @method  createRequestHandlers
+         *
+         * @return {Object} Request handler map
+         */
+        createRequestHandlers: function () {
+            return {
+                'MapModulePlugin.ToggleNavigationRequest': this
+            };
+        },
+        /**
+         * Handler for MapModulePlugin.ToggleNavigationRequest
+         *
+         * Oskari.getSandbox().postRequestByName('MapModulePlugin.ToggleNavigationRequest', [true/false]);
+         *
+         * @param {Oskari.mapframework.core.Core} core
+         *      Reference to the application core (reference sandbox core.getSandbox())
+         * @param {Oskari.mapframework.bundle.mapmodule.request.ToggleNavigationRequest} request
+         *      Request to handle
+         */
+        handleRequest: function (core, request) {
+            this._isVisible = request.isVisible();
+            this.refresh();
+        },
+        isVisible: function () {
+            return this._isVisible;
+        },
+        setState: function (state) {
+            this.state = state || {};
+            if (state.navigationVisible) {
+                this._hideNavigation();
+            } else {
+                this._showNavigation();
+            }
+            this.refresh();
+        },
+        getState: function () {
+            return this.state;
+        },
+        _showNavigation: function () {
+            const nav = [...Oskari.dom.getRootEl().children].find(c => c.localName === 'nav');
+            nav.style.display = 'block';
+        },
+        _hideNavigation: function () {
+            const nav = [...Oskari.dom.getRootEl().children].find(c => c.localName === 'nav');
+            nav.style.display = 'none';
+        }
```

Comment:
Let's move and combine these to functions under `Oskari.dom.showNavigation(boolean show = true)`. https://github.com/oskariorg/oskari-frontend/blob/master/src/oskari.dom.js